Signal/src/ViewControllers/ConversationView/ConversationHeaderView.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+public protocol ConversationHeaderViewDelegate {
+    func didTapConversationHeaderView(_ conversationHeaderView: ConversationHeaderView)
+}
+
+@objc
+public class ConversationHeaderView: UIStackView {
```

Comment:
I'm surprised this works.  I thought it was unsafe to rely on iOS auto layout in navigation headers, and I presume UIStackView uses iOS auto layout internally.  

Let's test:

* On iOS 9, 10, 11.
* scenarios like leaving and returning to this view (by pushing convo settings, for example).
* Changes to content (group name length changes).

Reply:
> I'm surprised this works. I thought it was unsafe to rely on iOS auto layout in navigation headers, and I presume UIStackView uses iOS auto layout internally.

Here's my understanding:

On older versions of iOS, items added to a navbar are in a weird situation - primarily what is their superview from which we can derive their frame? The inspector shows some privately generated thing that we never have direct access to, but apparently it doesn't play nice with autolayout constraints. 

On iOS11, instead of some weird private class, it's just a stack view.

So before iOS11, it was not possible to use autolayout to set the frame of the item we are adding to the navbar, but in iOS11 it is.

However, it was **always** safe (iOS9-11) to use autolayout to determine the frame of the **subviews**(in this case the two labels and the avatar) of the item added to the navbar.

This is why we manually set the frame size of the ConversationHeaderView on iOS9/10, but on iOS11 use automatic sizing (which the new iOS11 navbar stackview respects).

And since the item we add to the navbar now has a defined frame, its subviews are free to pin to their superview as we would elsewhere.